app/components/search_bar/search_bar.js

```diff
@@ -0,0 +1,390 @@
+                    },
+                ),
+                Animated.timing(
+                    this.searchContainerAnimated,
+                    {
+                        toValue: 0,
+                        duration: 200,
+                    },
+                ),
+            ]).start();
+            resolve();
+        });
+    }
+
+    render() {
+        const {backgroundColor, ...restOfInputPropStyles} = this.props.inputStyle;
+
+        let clearIcon = null;
+        let searchIcon = null;
+        let cancelIcon = null;
+
+        if (Platform.OS === 'ios') {
+            clearIcon = {
+                type: 'ionicon',
+                name: 'ios-close-circle',
+                size: 17,
+                color: this.props.tintColorDelete ? this.props.tintColorDelete : styles.defaultColor.color,
+            };
+
+            searchIcon = (
+                <EvilIcon
+                    name='search'
+                    size={24}
+                    style={[
+                        styles.fullWidth,
+                        this.props.tintColorSearch && {color: this.props.tintColorSearch},
+                        {
+                            top: middleHeight - 10,
+                        },
+                    ]}
+                />
+            );
+        } else {
+            searchIcon = this.props.showArrow ?
+                (
+                    <TouchableWithoutFeedback onPress={this.onCancel}>
+                        <MaterialIcon
+                            name='arrow-back'
+                            size={this.props.backArrowSize}
+                            color={this.props.titleCancelColor || this.props.placeholderTextColor}
+                        />
+                    </TouchableWithoutFeedback>
+                ) :
+                {
+                    type: 'material',
+                    size: this.props.searchIconSize,
+                    color: this.props.tintColorSearch || this.props.placeholderTextColor,
+                    name: 'search',
+                };
+
+            // Making sure the icon won't change depending on whether the input is in focus on Android devices
+            cancelIcon = searchIcon;
+
+            clearIcon = {
+                type: 'material',
+                size: this.props.deleteIconSize,
+                color: this.props.titleCancelColor || this.props.placeholderTextColor,
+                name: 'close',
+            };
+        }
+
+        return (
+            <View style={styles.container}>
+                {((this.props.leftComponent) ?
+                    <Animated.View
+                        style={{
+                            right: this.leftComponentAnimated,
+                        }}
+                        onLayout={this.onLeftComponentLayout}
+                    >
+                        {this.props.leftComponent}
+                    </Animated.View> :
+                    null
+                )}
+                <Animated.View
```

Comment:
Why is this Animated.View needed here ?

Reply:
Ah that was a remnant of a previous animation that is no longer running, I should have changed it to `View`

Reply:
Actually nvm, this particular element is animated when a `leftComponent` is present so that the input expands to fill the whole width of its parent when the `leftComponent` is animated out of the screen